### ISSUE TITLE
save not called in forms

### ISSUE THREAD

**Issue by RWOverdijk** [marmelab/admin-on-rest#855]
### Long version:

I have titles of movies, and those movies have annotations. Annotations have a title, description and image. That's a oneToMany relationship, except it's not a relationship but an embed (firebase). I want to be able to edit those annotations in the nested object. Example structure:

```json
{
  "settings": {
    "highlightColor": "#FFB001"
  },
  "video": {
    "title": "Main video title",
    "url": "http://foo-bar.amazonaws.com/video.mp4",
    "images": {
      "timeline": "https://to/poster/small-320x180.jpg",
      "sidebar": "https://to/poster/medium-680x360.jpg",
      "poster": "https://path.to/high-1280x720.jpg"
    }
  },
  "annotations": [
    {
      "type": "text",
      "start": 40000,
      "end": 48000,
      "title": "Text Title - Lorem ipsum dolor",
      "description": "Text Description - Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?",
      "source": "Text Source - Lorem ipsum dolor sit.",
      "url": "https://www.url.com"
    },
    {
      "type": "text",
      "start": 40000,
      "end": 48000,
      "title": "Text Title - Lorem ipsum dolor",
      "description": "Text Description - Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?",
      "source": "Text Source - Lorem ipsum dolor sit.",
      "url": "https://www.url.com"
    }
  ]
}
```

I'm using tabs to display the segments (settings, video, annotations). What I had in mind was a table of rows on the annotations tab, with a "create" button which opens a modal in which I can create the new annotation. Edit does the same, but with data filled out. All of this _without_ sending it to the server, until I press the save button in the toolbar.

### TL;DR; Version

Tab and simple form have a `save` prop. When defined, it's not being called.

What I expect to happen, is that I can handle the form submit (capture it, instead of sending it to the server) so I can do something with the result, instead of the default behavior.

**Comment by fzaninotto** [comment]
Nope, this `save` prop isn't documented because it's not public API. If you want to handle save yourself, use your own `Edit` component.

**Comment by RWOverdijk** [comment]
What does "use your own Edit component" mean? Any docs?

**Comment by fzaninotto** [comment]
Admin-on-rest is just a set of React components. When admin-on-rest components can't do what you want, use your own component as `edit` prop in the `<Resource>` component (see https://marmelab.com/admin-on-rest/Resource.html).

**Comment by RWOverdijk** [comment]
But I only want to enhance a single tab's behavior. Be able to manipulate the results. Like I said on another issue, I'm excited about this and want to contribute. But you seem a bit quick to dismiss my questions. Having a way to hook into the save behavior of a form is a tiny, but powerful feature that allows you to do a lot more with a lot less.

**Comment by fzaninotto** [comment]
I'm sorry if I sound too quick to dismiss your questions. It turns out it's question #855 and this project is very young. If we want to make it right, we need to stay focused on a limited amount of features. Please keep in mind that this is a voluntary project, and that nobody gets paid.

Therefore, we use this issue tracker to discuss bugs only. If you need support, I advise you to ask your question again on stack overflow:

https://stackoverflow.com/questions/tagged/admin-on-rest

**Comment by RWOverdijk** [comment]
I know what that's like. Just take a quick browse through [my projects](https://github.com/spoonx)... :) I'm not getting paid either haha.

That's fair, I'll point my arrows on stackoverflow. Thanks!
